cpp/src/gandiva/lru_cache.h

```diff
@@ -100,11 +136,58 @@ class LruCache {
     }
   }
 
+  arrow::util::optional<value_type> getObject(const key_type& key) {
+    // lookup value in the cache
+    typename map_type::iterator value_for_key = map_.find(key);
+
+    std::string obj_file_name = "obj-" + std::to_string(key.Hash()) + ".cache";
+    llvm::SmallString<128>obj_cache_file = cache_dir_;
+    llvm::sys::path::append(obj_cache_file, obj_file_name);
+
+    if (value_for_key == map_.end()) {
+      return arrow::util::nullopt;
+    }
+
+    // return the value, but first update its place in the most
+    // recently used list
+    typename list_type::iterator position_in_lru_list = value_for_key->second.second;
+    if (position_in_lru_list != lru_list_.begin()) {
+      // move item to the front of the most recently used list
+      lru_list_.erase(position_in_lru_list);
+      lru_list_.push_front(key);
+
+      // update iterator in map
+      position_in_lru_list = lru_list_.begin();
+      const value_type& value = value_for_key->second.first;
+      map_[key] = std::make_pair(value, position_in_lru_list);
+
+      // return the value
+      return value;
+    } else {
+      // the item is already at the front of the most recently
+      // used list so just return it
+      return value_for_key->second.first;
+    }
+  }
+
   void clear() {
     map_.clear();
     lru_list_.clear();
+    cache_size_ = 0;
   }
 
+  std::string toString(){
+    auto lru_size = lru_list_.size();
+    std::string string = "LRU Cache list size: " + std::to_string(lru_size) + "."
+                         + " LRU Cache size: " + std::to_string(cache_size_);
+    return string;
+  }
+
+  size_t getLruCacheSize(){
+    return cache_size_;
+  }
```

Comment:
Check the if the methods names should follow the pattern to start with capital letter

Reply:
I checked, and I might be the case, so I capitalized the first letter.